cypress/e2e/environment.cy.js

```diff
@@ -0,0 +1,389 @@
+    newValue: "375",
+  },
+  {
+    name: "protected_environment",
+    inputType: "switch",
+    newValue: "true",
+  },
+  {
+    name: "purge_on_delete",
+    inputType: "switch",
+    newValue: "true",
+  },
+  {
+    name: "push_on_auto_deploy",
+    inputType: "switch",
+    newValue: "false",
+  },
+  {
+    name: "resource_action_logs_retention",
+    inputType: "text",
+    newValue: "8",
+  },
+  {
+    name: "server_compile",
+    inputType: "switch",
+    newValue: "false",
+  },
+];
+const newEnvCard = '[aria-label="Overview-Success"] > :first-child';
+const projectNameInput = '[aria-label="Project Name-typeahead"]';
+const newProjectName = '[aria-label="Project Name-select-input';
+const newProjectNameButton = ".pf-c-select__menu-item";
+const nameInput = '[aria-label="Name-input"]';
+const descInput = '[aria-label="Description-input"]';
+const repoInput = '[aria-label="Repository-input"]';
+const branchInput = '[aria-label="Branch-input"]';
+const iconInput = "#simple-text-file-filename";
+
+const testProjectName = (number) => "Test Project Name " + number;
+const testName = (number) => "TestName " + number;
+
+const createEnv = (number, shouldTypeName = true, isShort = false) => {
+  cy.get(projectNameInput).type(testProjectName(number));
+  cy.get(newProjectName).within(() => {
+    cy.get(newProjectNameButton).click();
+  });
+  if (shouldTypeName) {
+    cy.get(nameInput).type(testName(number));
+  }
+  if (!isShort) {
+    cy.get(descInput).type("Test description");
+    cy.get(repoInput).type("repository");
+    cy.get(branchInput).type("branch");
+    cy.get(iconInput).attachFile(icon);
+  }
+};
+
+const deleteEnv = (number) => {
+  cy.get("button").contains("Delete environment").click();
+  cy.get('[aria-label="delete"]').should("be.disabled");
+
+  cy.get('[aria-label="delete environment check"]').type(testName(number));
+  cy.get('[aria-label="delete"]').should("not.be.disabled").click();
+  cy.url().should("eq", Cypress.config().baseUrl + "/console/");
+  cy.get('[aria-label="Environment card"]')
+    .contains(testProjectName(number))
+    .should("not.exist");
+};
+
+describe("Environment", function () {
+  it("1.1 cancel creation of an environment", function () {
+    cy.visit("/console/");
+    cy.get(newEnvCard).click();
+    cy.url().should(
+      "eq",
+      Cypress.config().baseUrl + "/console/environment/create"
+    );
+    createEnv(1);
+    cy.get("button").contains("Cancel").click();
+    cy.url().should("eq", Cypress.config().baseUrl + "/console/");
+  });
+
+  it("1.2 Create ne environment", function () {
+    //fill the form and submit
+    cy.visit("/console/environment/create");
+    createEnv(2, false);
+    cy.get("button").contains("Submit").should("be.disabled");
+    cy.get(nameInput).type(testName(2));
+    cy.get("button").contains("Submit").should("not.be.disabled");
+    cy.get("button").contains("Submit").click();
+    cy.url().should("contain", "/console/lsm/catalog?env=");
```

Comment:
This really depends on how much it will be used. Factoring this all out in constants is not going to make the code more readable. Especially in test cases I am actually against this. When this changes, you can also do a search and replace.

Reply:
It actually is being reused, and I believe having an object like so
```javascript
const PATHS = {
     create_env: "/console/lsm/catalog?env=",
     settings_env: "....",
```
would be useful

Reply:
This is similar to the previous case, 

Neither the argument `The url is or should ba a constant, can be extracted` nor `It actually is being reused,` 
carry any weight with me, because I don't see any intrinsic value in either. 

However, you do neatly demonstrate that there is an abstraction that can be extracted: the overview of paths. 

So here the question is harder for me to answer, because there are some delicate balances:
- reduced locality, increased tangling
- potential increase in readability (by naming URL's), potential decreased readability (by hiding urls)
- very subtle change in what you test: the url is exactly this Vs what is expected in general (would matter if we test for URL compatibility over versions, but we don't so it doesn't really matter)  
- easier to change routing (ridiculous argument, as I expect all hell to break loose anyways and this will be a drop in the sea)

Reply:
As agreed on in the meeting, this is a logic ensemble of values, but it doesn't bring that much benefit to store them in objects. Both solutions are fine for me. @matborowczyk you can resolve the thread and specify whichever solution you decided to pick, I'll keep in sync then with the tests I'm writing ;)